src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1234,9 +1234,22 @@ void EIRPlantLoopHeatPump::checkConcurrentOperation(EnergyPlusData &state)
             continue;
         }
         if (thisPLHP.running && thisPLHP.companionHeatPumpCoil->running) {
-            ShowRecurringWarningErrorAtEnd(state,
-                                           "Companion heat pump objects running concurrently, check operation.  Base object name: " + thisPLHP.name,
-                                           thisPLHP.recurringConcurrentOperationWarningIndex);
+            ShowSevereError(state, "Concurrent plant loop heat pump companion operation.");
+            ShowContinueErrorTimeStamp(state, "");
+            ShowContinueError(state, "PLHP name= " + thisPLHP.name);
+            ShowContinueError(state, format("PLHP source side inlet temperature= {:.2R} [C]", thisPLHP.sourceSideInletTemp));
+            ShowContinueError(state, format("PLHP source side outlet temperature= {:.2R} [C]", thisPLHP.sourceSideOutletTemp));
+            ShowContinueError(state, format("PLHP source side heat transfer rate= {:.2R} [W]", thisPLHP.sourceSideHeatTransfer));
+            ShowContinueError(state, "PLHP companion coil name= " + thisPLHP.companionHeatPumpCoil->name);
+            ShowContinueError(
+                state, format("PLHP companion coil source side inlet temperature= {:.2R} [C]", thisPLHP.companionHeatPumpCoil->sourceSideInletTemp));
+            ShowContinueError(
+                state,
+                format("PLHP companion coil source side outlet temperature= {:.2R} [C]", thisPLHP.companionHeatPumpCoil->sourceSideOutletTemp));
+            ShowContinueError(
+                state,
+                format("PLHP companion coil source side heat transfer rate= {:.2R} [W]", thisPLHP.companionHeatPumpCoil->sourceSideHeatTransfer));
+            ShowFatalError(state, "CheckConcurrentOperation: Simulation terminated because of problems in plant loop heat pump operation");
```

Comment:
I am not sure this is the correct method. I think this would occur when there is a cooling and heating HP on the same plant loop. I mean, this is a HP so there is only 1 HX. Ok, that makes sense to have a method to heat and cool the loop to be within some temperature range. If this is true, why would the inlet temperature of one HP ever be different from the other? Unless of course you have 2 HXs and one of those HPs is upstream of the other. And if that is the case then if the first one runs then the second one should be off. As an example, the plant loop temperature set points would be say 10C for cooling and something lower, like 7C for heating so that the plant loop temp stayed between 7C and 10C.. Did I get those control temps correct? So if the inlet temp = 11C, the cooling HP would be on and the heating HP off, even if the cooling HP operated the heating HP inlet temp should not be below 7C. Same if inlet temp = 6C, only heating HP would operate. So what scenario would cause both HPs to operate? Switched cooling/heating set point temperature? One HP overshooting the set point? What is happening in the defect file to cause both HPs to operate?